lib/lib/datamanagement/repository/grid_repository.dart

```diff
@@ -0,0 +1,45 @@
+import 'dart:convert';
+import 'package:f1_application/lib/model/driver.dart';
+import 'package:f1_application/lib/model/season.dart';
+import 'package:f1_application/lib/network/api/ergast_api_client.dart';
+import 'package:flutter/cupertino.dart';
+
+class GridRepository with ChangeNotifier {

```

Comment:
Itt a legnagyobb baj az, hogy ez nem egy Repository osztály, hanem egy ViewModel osztály, ami össze van mosva a Repository felelősségi körével. Ezt szét kéne szedni legalább két részre mindegyik képernyőnél: 

1, <Season/Grid/DriverProfile>ViewModel - itt annak az objektumnak/objektumoknak kellene lennie, amit a felületen meg akarunk majd jeleníteni, plusz olyan eseménykezelő függvényeknek, amiket a felületről hívni szeretnénk, pl. `fetch` kezdetű függvények teljesen jók itt
2, <Season/Grid/DriverProfile>Repository - ezeken semmi másnak nem kéne lenni, csak egy `async fetch...` függvénynek (és private API változóknak), ami visszaadja a szerializált objektumot a hívónak. Például a GridRepository-n így kéne kinéznie az interface-nek

```
Future<List<Driver>> fetchGrid(SeasonModel season) { async ... }
```

3, A kettő közé még szükség van a `Service` rétegre is, ezt bekommentelem külön a képernyő osztályára


https://developer.webstar.hu/mobil/code-review-szempontok/
> Nagy osztály: túl nagyra nőtt osztály 
> Magas szintű architektúrális minta követése